test/e2e/extra_test.go

```diff
@@ -0,0 +1,67 @@
+package e2e_test
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gbytes"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/actions"
+	kubecli "github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/cli/kubecli"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/model"
+	"github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/utils"
+)
+
+var _ = Describe("Deploy cluster", Label("cluster-extra-ns"), func() {
+	var data model.TestDataProvider
+
+	BeforeEach(func() {
+		Eventually(kubecli.GetVersionOutput()).Should(Say(K8sVersion))
+	})
+	AfterEach(func() {
+		GinkgoWriter.Write([]byte("\n"))
+		GinkgoWriter.Write([]byte("===============================================\n"))
+		GinkgoWriter.Write([]byte("Operator namespace: " + data.Resources.Namespace + "\n"))
+		GinkgoWriter.Write([]byte("===============================================\n"))
+		if CurrentSpecReport().Failed() {
+			GinkgoWriter.Write([]byte("Test has been failed. Trying to save logs...\n"))
+			utils.SaveToFile(
+				fmt.Sprintf("output/%s/operatorDecribe.txt", data.Resources.Namespace),
+				[]byte(kubecli.DescribeOperatorPod(data.Resources.Namespace)),
+			)
+			utils.SaveToFile(
+				fmt.Sprintf("output/%s/operator-logs.txt", data.Resources.Namespace),
+				kubecli.GetManagerLogs(data.Resources.Namespace),
+			)
+			actions.SaveTestAppLogs(data.Resources)
+			actions.SaveK8sResources(
+				[]string{"deploy", "atlasclusters", "atlasprojects"},
+				data.Resources.Namespace,
+			)
+			actions.AfterEachFinalCleanup([]model.TestDataProvider{data})
+		}
+	})
+
+	DescribeTable("Namespaced operators working only with its own namespace with different configuration",
+		func(test model.TestDataProvider) {
+			data = test
+			mainCycle(test)
+		},
+		Entry("Trial - Simplest configuration with no backup", Label("ns-crd"),
+			model.NewTestDataProvider(
```

Comment:
`model.NewTestDataProvider` is probably something from Atlas, but generally I think if you have more than 3 arguments you should just pass a struct :(